client/src/components/Auth/SignUp.jsx

```diff
@@ -1,108 +1,106 @@
+                axios
+                    .post(
+                        "https://wheretocode-master.herokuapp.com/auth/register",
+                        newUser
+                    )
+                    .then(res => {
+                        console.log(res.data);
+                        setUsername("");
+                        setEmail("");
+                        setPasswordOne("");
+                        props.history.push(ROUTES.HOME);
+                    })
+                    .catch(error => {
+                        console.log(error);
+                    });
             })
             .catch(error => {
-                this.setState({ error });
+                setError(error);
             });
-
         event.preventDefault();
     };
 
-    onChange = event => {
-        this.setState({ [event.target.name]: event.target.value });
-    };
-
-    render() {
-        const {
-            username,
-            email,
-            passwordOne,
-            passwordTwo,
-            error,
-        } = this.state;
-
-        const isInvalid =
-            passwordOne !== passwordTwo ||
-            passwordOne === '' ||
-            email === '' ||
-            username === '';
+    const isInvalid =
+        passwordOne !== passwordTwo ||
+        passwordOne === "" ||
+        email === "" ||
+        username === "";
 
-        return (
-            <Box width="medium">
-            <Form onSubmit={this.onSubmit}>
+    return (
+        <Box width="medium">
+            <Form onSubmit={onSubmit}>
                 <FormField
                     name="username"
                     value={username}
-                    onChange={this.onChange}
+                    onChange={e => setUsername(e.target.value)}
                     type="text"
                     placeholder="Full Name"
                 />
                 <FormField
                     name="email"
                     value={email}
-                    onChange={this.onChange}
+                    onChange={e => setEmail(e.target.value)}
                     type="text"
                     placeholder="Email Address"
                 />
                 <FormField
                     name="passwordOne"
                     value={passwordOne}
-                    onChange={this.onChange}
+                    onChange={e => setPasswordOne(e.target.value)}
                     type="password"
                     placeholder="Password"
                 />
                 <FormField
                     name="passwordTwo"
                     value={passwordTwo}
-                    onChange={this.onChange}
+                    onChange={e => setPasswordTwo(e.target.value)}
                     type="password"
                     placeholder="Confirm Password"
                 />
                 <Button disabled={isInvalid} type="submit" primary label="Sign Up" />
```

Comment:
I like your implementation of what I assume to be a built in prop of disabled in grommet?

Reply:
Disabled is an HTML button attribute but Grommet makes it look really good. 

I really like the way it's used here by setting the parameters so the button is invalid until all of the required information is there. 👍